core/src/subgraph/instance_manager.rs

```diff
@@ -355,7 +352,21 @@ where
                 &network,
                 &required_capabilities, e))?.clone();
 
-        store.start_subgraph_deployment(&logger, &manifest.id)?;
+        {
```

Comment:
I think this is ok. In [this commit](https://github.com/graphprotocol/graph-node/pull/2313/commits/55642e78d82c124703c1109fa8bacc774f34b284) I had to move `start_subgraph_deployment` to the same thread that does `run_subgraph` since, with copying, `start_subgraph_deployment` can take hours and would otherwise block the instance manager. (There's nothing to do for this PR, just an FYI)

Reply:
Good thing it is in a blocking task then! Ideally it would use something like `with_conn` but it didn't seem so straightforward to refactor it, so that's left for another day.